### PR TITLE
update Gen3File to include upload_file and delete_files

### PR DESCRIPTION
**Pull request by plooploops**
### New Features
Add `upload_file` and `delete_files` to `Gen3File`

### Improvements
Update test suite for `Gen3File` coverage

### Dependency updates
Include `pytest-cov = "^2.12.1"`

### CODE REVIEW

File: tests/test_file.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+            file_name="file.txt",
+            authz=authz,
+            protocol=supported_protocol,
+            expires_in=expires_in,
+        )
+        if status_code == 201:
+            # check that the response URL includes the extra guid for the uploaded file
+            assert res.get("url") == expected_response["url"].replace(
+                f"{expected_response['guid']}/", ""
+            )
```
Comment (paulineribeyre) [inline tests/test_file.py:263]: i don't understand what real functionality this (replacing the guid in the url) is testing? all this data is mocked

**Review by paulineribeyre — COMMENTED** [review]
Thank you for contributing! a couple small comments below

File: gen3/file.py [hunk 2]
```diff
@@ -47,8 +47,59 @@ def get_presigned_url(self, guid, protocol=None):
...
             api_url += "?protocol={}".format(protocol)
         output = requests.get(api_url, auth=self._auth_provider).text
+
+        try:
+            data = json.loads(output)
+        except:
+            return output
+        return data
+
+    def delete_files(self, guid):
```
[the remark this answers is not in the document]
Comment (plooploops) [reply gen3/file.py]: Sounds good, did a rename!

File: gen3/file.py [hunk 3]
```diff
@@ -47,8 +47,59 @@ def get_presigned_url(self, guid, protocol=None):
...
+        api_url = "{}/user/data/{}".format(self._endpoint, guid)
+        output = requests.delete(api_url, auth=self._auth_provider).text
+
+        return output
+
+    def upload_file(self, file_name, authz=None, protocol=None, expires_in=None):
+        """
+        Get a presigned url for a file to upload
+
+        :param str file_name:
```
[the remark this answers is not in the document]
Comment (plooploops) [reply gen3/file.py]: Updated, thanks for the pointer!

File: tests/test_file.py (hunk 1 above)
Comment (plooploops) [reply tests/test_file.py:263]: Based on earlier testing, it appeared that we were getting a url back like `https://somecontainer/some/path/guid/file.txt` when you call upload_file.

If this is no longer the case, we can simplify and just look for the a result like `https://somecontainer/some/path/file.txt` instead?

Comment (plooploops) [reply tests/test_file.py]: Simplifying the call, thanks for the insight here!

**Review by paulineribeyre — APPROVED** [review]
Tests passed in #104

**Comment by paulineribeyre** [comment]
Thank you @plooploops!

**Comment by plooploops** [comment]
> Thank you @plooploops!

Thanks for your patience and support! :)

**Pull request merged**